### PR TITLE
Database migration script to update old score overrides to match upstream

### PR DESCRIPTION
**Pull request by kluo**
`./scripts/migrate_score_override.py`

@dcadams @stvstnfrd @caesar2164 @caseylitton

### CODE REVIEW

File: scripts/migrate_score_override.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
+import itertools
+from django.contrib.auth.models import User
+from django.utils.timezone import now
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from openassessment.assessment.models import Assessment, AssessmentPart, StaffWorkflow
+from openassessment.workflow.models import AssessmentWorkflow, AssessmentWorkflowStep
+from student.models import anonymous_id_for_user
+from submissions.models import Score, ScoreAnnotation, Submission
+
+old_scores = Score.objects.filter(submission__isnull=True, reset=False)
```
Comment (stvstnfrd) [inline scripts/migrate_score_override.py]: Scripts should generally not contain the logic directly in the body, but rather, should take a form like:

```python
def main():
    # do stuff here

if __name__ == '__main__':
    main()
```

The idea is that if another file were to import this file,
as is, the data migration would be run automatically.
Whereas, we only want to run this if the script is executed directly,
not if it's just imported.

I don't know how/if this is affected by the `execfile` call,
but we should consider using this idiom.

File: scripts/migrate_score_override.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    new_score = Score(
+        student_item=score.student_item,
+        submission=submission,
+        points_earned=points_override,
+        points_possible=points_possible,
+    )
+    new_score.save()
+
+    # Fetch staff id from score course for ScoreAnnotation
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(score.student_item.course_id)
```
Comment (dcadams) [inline scripts/migrate_score_override.py]: SlashSeparatedCourseKey has been deprecated by upstream albeit ahead of our merge point. ( https://github.com/edx/edx-platform/pull/15617 )

Maybe you can future-proof this with something like:

from opaque_keys.edx.keys import CourseKey
course_id = CourseKey.from_string(score.student_item.course_id)

File: scripts/migrate_score_override.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+old_scores = Score.objects.filter(submission__isnull=True, reset=False)
+for score in old_scores:
+    points_override = score.points_earned
+    points_possible = score.points_possible
+
+    # Update old override with new reset
+    score.points_earned = 0
+    score.points_possible = 0
+    score.reset = True
+    score.save()
```
Comment (dcadams) [inline scripts/migrate_score_override.py]: Doing this block here means if something goes wrong further down then the program cannot be run again and data has been lost.

File: scripts/migrate_score_override.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        points_possible=points_possible,
+    )
+    new_score.save()
+
+    # Fetch staff id from score course for ScoreAnnotation
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(score.student_item.course_id)
+    staff = User.objects.filter(
+        courseaccessrole__role='instructor',
+        courseaccessrole__course_id=course_id,
+    )[:1].get()
```
Comment (dcadams) [inline scripts/migrate_score_override.py]: Is there at least one 'instructor' in each of these courses or will it also select 'staff'?

File: scripts/migrate_score_override.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        print "CANNOT UPDATE ORA2 TABLES, NO EXISTING ASSESSMENTS FOR SUBMISSION " + submission.uuid
+        continue
+
+    rubric = assessment.rubric
+
+    staff_assessment = Assessment.create(
+        rubric=rubric,
+        scorer_id=staff_id,
+        submission_uuid=submission.uuid,
+        score_type="ST",
```
Comment (dcadams) [inline scripts/migrate_score_override.py]: Any idea of the meaning of 'ST'?

Comment (kluo) [reply scripts/migrate_score_override.py]: Short for staff

**Comment by dcadams** [comment]
:+1: 

pending tests.

**Comment by caesar2164** [comment]
test this please

**Comment by kluo** [comment]
test this please

**Comment by kluo** [comment]
test this please

**Pull request merged**